apps/web/test/utils/bookingScenario/MockPaymentService.ts

```diff
@@ -0,0 +1,88 @@
+import prismaMock from "../../../../../tests/libs/__mocks__/prisma";
+
+import type { Payment, Prisma, PaymentOption, Booking } from "@prisma/client";
+import { v4 as uuidv4 } from "uuid";
+import "vitest-fetch-mock";
+
+import { sendAwaitingPaymentEmail } from "@calcom/emails";
+import logger from "@calcom/lib/logger";
+import type { CalendarEvent } from "@calcom/types/Calendar";
+import type { IAbstractPaymentService } from "@calcom/types/PaymentService";
+
+export function getMockPaymentService() {
+  function createPaymentLink(/*{ paymentUid, name, email, date }*/) {
+    return "http://mock-payment.example.com/";
+  }
+  const paymentUid = uuidv4();
+  const externalId = uuidv4();
+
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  class MockPaymentService implements IAbstractPaymentService {
+    // TODO: We shouldn't need to implement adding a row to Payment table but that's a requirement right now.
+    // We should actually delegate table creation to the core app. Here, only the payment app specific logic should come
```

Comment:
Agree with this, we should have a payment core. Or maybe it will be too much abstraction.. :/

Reply:
I don't think that would be too much. It would make sense. We should really focus on such abstractions so that App authors need to right less and less code and thus can introduce less bugs.